### PR TITLE
fix(vscode): harden codex picker command-line quoting

### PR DESCRIPTION
**Pull request by Jacobinwwey**
### 1. Context & Motivation (Context)
- **Issue:** n/a
- **Problem:** VS Code command invocation mixed argument resolution and shell command synthesis inline, which made Linux/Unix quoting edge cases brittle (especially single quotes and workspace placeholder expansion).
- **Trade-offs:** Extracting command-line logic into a pure helper module increases files slightly, but gives deterministic unit-test coverage and lowers regression risk across shells.

### 2. Implementation Details (Implementation)
- Extracted VS Code shell command logic into `editors/vscode/src/commandLine.ts` with pure helpers for argument resolution, token quoting, and shell-specific close-on-success behavior.
- Updated `extension.ts` to use the helper module and pass explicit shell path into terminal command rendering.
- Added Node test coverage in `commandLine.test.ts` for quote escaping, workspace placeholder expansion, and shell close behavior.
- Added a dedicated `pnpm run test` script for the VS Code extension package.
- **Note:** This PR intentionally avoids touching Rust CLI behavior and does not change picker/session semantics.

### 3. Testing Strategies (Validation)
- [x] Added Unit Tests for `<editors/vscode/src/commandLine.ts>` edge cases (single quote escaping, placeholder expansion, shell close command).
- [x] Ran Rust workspace checks: `cargo fmt --all --check`, `cargo clippy --workspace --all-targets -- -D warnings`, `cargo test --workspace`.
- [ ] Ran VS Code package test command locally (`pnpm -C editors/vscode run test`)
- Local blocker: workspace `editors/vscode/node_modules` missing (`tsc: not found`) in this environment.
- [x] Verified backward compatibility for command defaults and extension command wiring.
- **Benchmark:** n/a

### CODE REVIEW

File: editors/vscode/src/commandLine.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  return `${commandLine}; code=$?; if [ "$code" -eq 0 ]; then exit 0; fi`;
+}
+
+export function quoteShellToken(value: string): string {
+  if (/^[A-Za-z0-9_./:@%+=,-]+$/.test(value)) {
+    return value;
+  }
+
+  return `'${value.replace(/'/g, `'\\''`)}'`;
+}
```
[the remark this answers is not in the document]
Comment (Jacobinwwey) [reply editors/vscode/src/commandLine.ts]: Fixed in 71287e8. `quoteShellToken` is now shell-aware for POSIX/PowerShell/CMD, and shell-name detection now handles both POSIX and Windows-style shell paths. Added regression tests covering PowerShell and CMD quoting.

File: editors/vscode/src/commandLine.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    if ((arg === "." || arg === "${workspaceFolder}") && args[index - 1] === "--cwd") {
+      return cwd;
+    }
+    return arg;
+  });
+}
+
+export function buildCommandLine(command: string, args: string[]): string {
+  return [command, ...args].map(quoteShellToken).join(" ");
+}
```
[the remark this answers is not in the document]
Comment (Jacobinwwey) [reply editors/vscode/src/commandLine.ts]: Fixed in 71287e8. `buildCommandLine` now accepts an optional `shellPath` and passes it through to `quoteShellToken`, so the same call path now supports shell-specific quoting end to end.

File: editors/vscode/package.json [hunk 3]
```diff
@@ -80,6 +78,7 @@
   "scripts": {
     "compile": "tsc -p ./",
     "watch": "tsc -watch -p ./",
+    "test": "pnpm run compile && rm -f out/extension.test.js && node --test out/commandLine.test.js",
```
[the remark this answers is not in the document]
Comment (Jacobinwwey) [reply editors/vscode/package.json]: Fixed in 71287e8. Removed the non-portable `rm -f` from the VS Code `test` script and now run `pnpm run compile && node --test out/commandLine.test.js` directly.

File: editors/vscode/src/extension.ts [hunk 4]
```diff
@@ -60,7 +61,11 @@ async function pickAgent(): Promise<void> {
   const terminal = getTerminal(terminalName, workspaceFolder.uri.fsPath, reuseTerminal);
   terminal.show();
   terminal.sendText(
-    buildTerminalCommandLine(buildCommandLine(command, args), closeTerminalOnSuccess),
+    buildTerminalCommandLine(
+      buildCommandLine(command, args),
```
[the remark this answers is not in the document]
Comment (Jacobinwwey) [reply editors/vscode/src/extension.ts]: Fixed in 71287e8. Updated extension.ts to pass vscode.env.shell into buildCommandLine so argument quoting matches the active VS Code shell.

**Comment by Jacobinwwey** [comment]
Addressed all open review threads in commit 71287e8 and pushed to branch `fix/vscode-codex-picker-commandline-v2`.

What changed:
- shell-aware argument quoting in `quoteShellToken` for POSIX, PowerShell, and CMD
- `buildCommandLine` now accepts `shellPath` and passes it through to the quoting function
- VS Code extension now passes `vscode.env.shell` into `buildCommandLine`
- VS Code `test` script is now cross-platform: `pnpm run compile && node --test out/commandLine.test.js` (removed `rm -f`)
- added regression tests for PowerShell/CMD quoting and shell-path detection with Windows-style separators

Validation run locally:
- `cd editors/vscode && pnpm test`
- `cargo fmt --all --check`
- `cargo clippy --workspace --all-targets -- -D warnings`
- `cargo test --workspace`

**Comment by Jacobinwwey** [comment]
Superseded by the split review-focused PR #13 (same VS Code quoting scope with updated tests and shell handling). Closing this older branch to keep review history clean.
